discovery-provider/src/tasks/index_eth.py

```diff
@@ -0,0 +1,104 @@
+import time
+import logging
+from web3 import Web3
+from web3.providers.rpc import HTTPProvider
+from src.tasks.celery_app import celery
+from src.utils.config import shared_config
+from src.eth_indexing.event_scanner import EventScanner
+from src.utils.helpers import load_eth_abi_values
+from src.tasks.cache_user_balance import get_token_address
+
+logger = logging.getLogger(__name__)
+
+CHAIN_REORG_SAFETY_BLOCKS = 10
+
+provider = HTTPProvider(shared_config["web3"]["eth_provider_url"])
+
+# Remove the default JSON-RPC retry middleware
+# as it correctly cannot handle eth_getLogs block range
+# throttle down.
+provider.middlewares.clear()  # type: ignore
+
+web3 = Web3(provider)
+
+# Prepare stub ERC-20 contract object
+eth_abi_values = load_eth_abi_values()
+ERC20 = web3.eth.contract(abi=eth_abi_values["AudiusToken"]["abi"])
+
+AUDIO_CHECKSUM_ADDRESS = get_token_address(web3, shared_config)
+
+
+def index_eth_transfer_events(db, redis):
+    scanner = EventScanner(
+        db=db,
+        redis=redis,
+        web3=web3,
+        contract=ERC20,
+        event_type=ERC20.events.Transfer,
+        filters={"address": AUDIO_CHECKSUM_ADDRESS},
+    )
+    scanner.restore()
+
+    # Assume we might have scanned the blocks all the way to the last Ethereum block
+    # that mined a few seconds before the previous scan run ended.
+    # Because there might have been a minor Ethereum chain reorganisations
+    # since the last scan ended, we need to discard
+    # the last few blocks from the previous scan results.
+    # Scan from [last block scanned] - [latest ethereum block]
+    # (with a potentially offset from the tail to attempt to avoid blocks not mined yet)
+    since_block = scanner.get_last_scanned_block() - CHAIN_REORG_SAFETY_BLOCKS
+
+    # Note that our chain reorg safety blocks cannot go negative
+    start_block = max(since_block, 0)
+    end_block = scanner.get_suggested_scan_end_block()
+    if start_block > end_block:
+        return
+
+    logger.info(f"Scanning events from blocks {start_block} - {end_block}")
+    start = time.time()
+
+    # Run the scan
+    result, total_chunks_scanned = scanner.scan(start_block, end_block)
+
+    logger.info(
+        "Reached end block for eth transfer events... saving events to database"
+    )
+    scanner.save(end_block)
+    duration = time.time() - start
+    logger.info(
+        f"Scanned total {len(result)} Transfer events, in {duration} seconds, \
+            total {total_chunks_scanned} chunk scans performed"
+    )
+
+
+@celery.task(name="index_eth", bind=True)
+def index_eth(self):
+    # Index AUDIO Transfer events to update user balances
+    db = index_eth.db
+    redis = index_eth.redis
+
+    # Define lock acquired boolean
+    have_lock = False
+
+    # Define redis lock object
+    update_lock = redis.lock("index_eth_lock", blocking_timeout=25)
```

Comment:
don't need a blocking_timeout since we pass in blocking=False to acquire, this doesn't do anything. having some long timeout for the lock is probably a good idea though

Reply:
Ah, interesting. I think blocking=False is still fine because that just means we'll fail to acquire the lock and that run through the job will fail. If the first run may take hours, then yeah I'm in favor of not including a timeout in the lock.